core/src/com/unciv/models/ruleset/unique/Conditionals.kt

```diff
@@ -246,6 +246,8 @@ object Conditionals {
             UniqueType.ConditionalHasNotUsedOtherActions ->
                 state.unit == null || // So we get the action as a valid action in BaseUnit.hasUnique()
                     state.unit.abilityToTimesUsed.isEmpty()
+            UniqueType.ConditionalStackedWithUnit -> state.relevantUnit != null && 
+                    state.relevantUnit!!.getTile().getUnits().any { it.matchesFilter(conditional.params[0]) && it != state.relevantUnit }
```

Comment:
Switch order of && - first check equality only then matches()
This unit is always in its own tile, plus equality check is faster